File: php/import_fancyfoods_to_db.php
<?php
// Fancy Foods CSV Import to MySQL Products Table
// CONFIGURE THESE
$db_host = 'localhost';
$db_user = 'root';
$db_pass = '';
$db_name = 'orders';
$csv_path = __DIR__ . '/../data/excel_price_sheets/fancyfoods/fancyfoods_extracted.csv';
$vendor = 'Fancy Foods';

$conn = new mysqli($db_host, $db_user, $db_pass, $db_name);
if ($conn->connect_error) {
    die('Connection failed: ' . $conn->connect_error);
}
$conn->set_charset('utf8');

function get_uom_id($conn, $uom_code) {
    $uom_code = strtoupper(trim($uom_code));
    if ($uom_code === '') $uom_code = 'LB';
    $stmt = $conn->prepare('SELECT id FROM uom WHERE code = ?');
    $stmt->bind_param('s', $uom_code);
    $stmt->execute();
    $stmt->bind_result($id);
    if ($stmt->fetch()) {
        $stmt->close();
        return $id;
    }
    $stmt->close();
    $stmt = $conn->prepare('INSERT INTO uom (code, name, description) VALUES (?, ?, ?)');
    $stmt->bind_param('sss', $uom_code, $uom_code, $uom_code);
    $stmt->execute();
    $new_id = $stmt->insert_id;
    $stmt->close();
    return $new_id;
}

function upsert_product($conn, $row, $uom_id, $vendor) {
    $price = is_numeric($row['Price']) ? $row['Price'] : 0.00;
    $stmt = $conn->prepare('INSERT INTO products (item_code, description, price, vendor, category, subcategory, uom_id, is_active) VALUES (?, ?, ?, ?, ?, ?, ?, TRUE)
        ON DUPLICATE KEY UPDATE description = VALUES(description), price = VALUES(price), category = VALUES(category), subcategory = VALUES(subcategory), uom_id = VALUES(uom_id), is_active = TRUE');
    $stmt->bind_param('ssdsssi', $row['Item Code'], $row['Description'], $price, $vendor, $row['Category'], $row['Subcategory'], $uom_id);
    $stmt->execute();
    $stmt->close();
}

if (($handle = fopen($csv_path, 'r')) !== FALSE) {
    $header = fgetcsv($handle);
    while (($data = fgetcsv($handle)) !== FALSE) {
        $row = array_combine($header, $data);
        $uom_id = get_uom_id($conn, $row['UOM']);
        upsert_product($conn, $row, $uom_id, $vendor);
    }
    fclose($handle);
    echo "Import complete.\n";
} else {
    echo "Failed to open CSV file.\n";
}
$conn->close();
?>

File: python/import_fancyfoods_to_db.py
import csv
import mysql.connector
from mysql.connector import Error

# CONFIGURE THESE
DB_CONFIG = {
    'host': 'localhost',
    'user': 'root',
    'password': '',  # Set your MySQL root password
    'database': 'orders',
}
CSV_PATH = r'd:/xampp/htdocs/order_system/data/excel_price_sheets/fancyfoods/fancyfoods_extracted.csv'
VENDOR = 'Fancy Foods'

# Helper: get uom_id, insert if not exists
def get_uom_id(cursor, uom_code):
    cursor.execute("SELECT id FROM uom WHERE code = %s", (uom_code,))
    result = cursor.fetchone()
    if result:
        return result[0]
    cursor.execute("INSERT INTO uom (code, name, description) VALUES (%s, %s, %s)", (uom_code, uom_code, uom_code))
    return cursor.lastrowid

# Helper: upsert product
def upsert_product(cursor, row, uom_id):
    cursor.execute("""
        INSERT INTO products (item_code, description, price, vendor, category, uom_id, is_active)
        VALUES (%s, %s, %s, %s, %s, %s, TRUE)
        ON DUPLICATE KEY UPDATE
            description = VALUES(description),
            price = VALUES(price),
            category = VALUES(category),
            uom_id = VALUES(uom_id),
            is_active = TRUE
    """, (
        row['Item Code'],
        row['Description'],
        row['Price'] if row['Price'] else 0.00,
        VENDOR,
        row['Category'],
        uom_id
    ))

def main():
    try:
        conn = mysql.connector.connect(**DB_CONFIG)
        cursor = conn.cursor()
        with open(CSV_PATH, newline='', encoding='utf-8') as csvfile:
            reader = csv.DictReader(csvfile)
            for row in reader:
                uom_code = row['UOM'].strip().upper() if row['UOM'] else 'LB'
                uom_id = get_uom_id(cursor, uom_code)
                upsert_product(cursor, row, uom_id)
        conn.commit()
        print('Import complete.')
    except Error as e:
        print(f'Error: {e}')
    finally:
        if 'conn' in locals() and conn.is_connected():
            cursor.close()
            conn.close()

if __name__ == '__main__':
    main()
